cpp/src/IceGrid/AdminSessionI.cpp

```diff
@@ -152,14 +150,12 @@ AdminSessionI::setObservers(
         throw Ice::ObjectNotExistException(__FILE__, __LINE__, current.id, "", "");
     }
 
-    const auto timeout = secondsToInt(_timeout);
-    assert(timeout != 0);
     const auto locator = _registry->getLocator();
     if (registryObserver)
     {
         setupObserverSubscription(
             TopicName::RegistryObserver,
-            addForwarder(registryObserver->ice_timeout(timeout)->ice_locator(locator)));
```

Comment:
I can't remember exactly but my guess is that this code was to ensure the connection timeout was always overridden to the session timeout value... This is no longer useful since connection timeouts are gone.